### ISSUE TITLE
How to obtain aleatoric uncertainty? 

### ISSUE THREAD

**Issue by bangxiangyong** [google/neural-tangents#188]
I am aware that the default inference implemented is based on Mean squared error (MSE) loss. Is there an implemented example or a way to obtain aleatoric uncertainty instead (either homoscedestic or heteroscedestic)? i.e. learning to output the variance of an isotropic Gaussian distribution.

**Comment by bangxiangyong** [comment]
Thanks for the reply!

I may be limited by knowledge; but wouldn't an MSE loss only capture epistemic uncertainty i.e. uncertainty about the possible models instead of uncertainty within the data?

I have knowledge about Bayesian neural networks (BNN) and am trying to draw the parallel to NNGP/NTK inferences when it comes to estimating the aleatoric uncertainty and was expecting something along the lines of training under NLL of Gaussian loss (instead of MSE). To estimate aleatoric uncertainty the BNN architecture has dual outputs in the heteroscedestic setup (one for mean and one for variance of the Gaussian); whereas in the homoscedestic setup a free parameter is used for estimating the Gaussian variance. I guess one approach would be modifying the loss function of the NNGP to a Gaussian NLL (instead of MSE); However, i fail to find an example that does so. For references on estimating aleatoric uncertainty I am referring to setups such as the ones below:

[https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0252108](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0252108)
 (Equation 12)

[https://proceedings.neurips.cc/paper_files/paper/2017/file/2650d6089a6d640c5e85b2b88265dc2b-Paper.pdf](https://proceedings.neurips.cc/paper_files/paper/2017/file/2650d6089a6d640c5e85b2b88265dc2b-Paper.pdf)
 (Equation 5)

I hope i am making some sense!..
